compiler/rustc_target/src/spec/tests/tests_impl.rs

```diff
@@ -143,8 +143,9 @@ impl Target {
             assert!(self.executables);
         }
 
-        // Check crt static stuff
-        if self.crt_static_default || self.crt_static_allows_dylibs {
+        // Check crt static stuff. Trusty only supports crt static and so does not enable
+        // `crt_static_respected`.
+        if (self.crt_static_default || self.crt_static_allows_dylibs) && self.os != "trusty" {
```

Comment:
How do other crt static only targets handle this?

Reply:
As far as I can tell, it looks like the other targets that set `crt_static_default: true` are also setting `crt_static_respected: true`. I'm not sure what other targets do crt static-only the way we do for Trusty, but based on the consistency check they'd all have to be setting `crt_static_respected: true`.

Reply:
Wasm32-wasi is an example of a target that only supports static linking of libc.

Reply:
Yup, and that target is using `crt_static_respected: true`. Should I switch back to doing it that way and revert the change to the test? That's how I originally had it, so I at least know that that configuration works for how we've been building Trusty.

Reply:
I guess.